tools/cldr-apps/src/main/java/org/unicode/cldr/web/api/VoteAPIHelper.java

```diff
@@ -185,9 +185,8 @@ public static RowResponse getRowsResponse(
         String xp = null;
 
         if (args.xpstrid == null && args.page != null) {
-            try {
-                pageId = PageId.valueOf(args.page);
-            } catch (IllegalArgumentException iae) {
+            pageId = PageId.forString(args.page);
+            if (pageId == null) {
```

Comment:
I guess this returns null if it catches the exception?

```
        public static PageId forString(String name) {
            try {
                return PageIdNames.forString(name);
            } catch (Exception e) {
                throw new ICUException("No PageId for " + name, e);
            }
        }
```

Reply:
that's a feature of Java I wasn't familiar with

Reply:
Oh, right; I misread it. It catches IllegalArgumentException then throws ICUException.

So the removal of `catch (Throwable t)` means that searchPathheader may throw ICUException where before it wouldn't have. That goes beyond merely changing valueOf to forString. Here `q` comes from an http request, so it could be anything... Line 765 of SurveyAjax will catch it:

```
        } catch (Throwable e) {
            SurveyLog.logException(
                    logger, e, "Error in SurveyAjax?what=" + what + ": " + e.getMessage());
            sendError(out, e);
        }
```

I guess that's OK, though shouldn't searchPathheader then be declared to throw ICUException? It still silently catches other exceptions further below; it just seems a little arbitrary...